core/src/avm2/globals/flash/text/engine/GroupElement.as

```diff
@@ -24,18 +32,22 @@ package flash.text.engine {
         }
 
         public function replaceElements(beginIndex:int, endIndex:int, newElements:Vector.<ContentElement>):Vector.<ContentElement> {
-            if (beginIndex == endIndex && newElements.length == 1) {
-                if (beginIndex == this._elements.length) {
-                    this._elements.push(newElements[0]);
-                } else if (beginIndex < this._elements.length) {
-                    this._elements[beginIndex] = newElements[0];
-                } else {
-                    throw new RangeError("Error #2006: The supplied index is out of bounds.", 2006);
+            var old = this._elements.AS3::splice(beginIndex, endIndex - beginIndex);
+            if (newElements) {
+                for (var i = 0; i < newElements.length; i++) {
+                    this._elements.AS3::insertAt(beginIndex + i, newElements[i]);
                 }
-            } else {
-                stub_method("flash.text.engine.GroupElement", "replaceElements", "with beginIndex != endIndex or newElements.length != 1");
-                return new Vector.<ContentElement>();
             }
+            return old;
+        }
+
+        public function splitTextElement(elementIndex:int, splitIndex:int): TextElement {
+            var textElement: TextElement = this._elements[elementIndex];
```

Comment:
What happens here if `this._elements[elementIndex]` is not a `TextElement`? Does Flash also throw a coercion failure?

Reply:
Flash seems to throw:
```
ArgumentError: Error #2004: One of the parameters is invalid.
```

Reply:
Can this also throw in that case?